File: src/main/java/org/jukeboxmc/potion/SpeedEffect.java
package org.jukeboxmc.potion;

import org.jetbrains.annotations.NotNull;
import org.jukeboxmc.entity.EntityLiving;
import org.jukeboxmc.entity.attribute.AttributeType;

import java.awt.*;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class SpeedEffect extends Effect {

    @Override
    public int getId() {
        return 1;
    }

    @Override
    public @NotNull EffectType getEffectType() {
        return EffectType.SPEED;
    }

    @Override
    public @NotNull Color getEffectColor() {
        return new Color( 124, 175, 198 );
    }

    @Override
    public void update( long currentTick ) {

    }

    @Override
    public void apply(@NotNull EntityLiving entityLiving ) {
        entityLiving.setMovement( ( this.amplifier + 1 ) * 0.2f );
    }

    @Override
    public void remove(@NotNull EntityLiving entityLiving ) {
        entityLiving.getAttribute( AttributeType.MOVEMENT ).reset();
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/effect/SpeedEffect.kt
package org.jukeboxmc.server.effect

import org.jukeboxmc.api.effect.EffectType
import org.jukeboxmc.server.entity.Attribute
import org.jukeboxmc.server.entity.JukeboxEntityLiving
import java.awt.Color

class SpeedEffect : JukeboxEffect() {

    override fun getId(): Int {
        return 1
    }

    override fun getEffectType(): EffectType {
        return EffectType.SPEED
    }

    override fun getEffectColor(): Color {
        return Color(124, 175, 198)
    }

    override fun apply(entityLiving: JukeboxEntityLiving) {
        entityLiving.setMovement((this.getAmplifier() + 1) * 0.2f)
    }

    override fun update(currentTick: Long) {

    }

    override fun remove(entityLiving: JukeboxEntityLiving) {
        entityLiving.getAttribute(Attribute.MOVEMENT).reset()
    }
}